core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java

```diff
@@ -219,8 +219,16 @@ public String startWorkflow(String workflowId, String name, int version, Map<Str
 			// send wf start message
 			notifyWorkflowStatus(wf, StartEndState.start);
 
-			decide(workflowId);
-			logger.debug("Workflow has started.Current status=" + wf.getStatus() + ",workflowId=" + wf.getWorkflowId()+",CorrelationId=" + wf.getCorrelationId()+",input="+wf.getInput());
+			// Do not run decider right away
+			// The decider might block the thread until some task of the sub workflow
+			// (and their sub workflow, and so on) releases it
+			if (lazyDecider) {
```

Comment:
This logic is reused several times. We should consider extracting it into a method.